docs/quickstart-upgrade.md

```diff
@@ -0,0 +1,121 @@
+  if (navigationAction == "watch" && classId) {
+    targetUrl.pathname = "/hostSitePath/watch";
+    targetUrl.searchParams.set("class-id", classId ?? "");
+  } else if (navigationAction == "discover") {
+    targetUrl.pathname = "/hostSitePath/discover";
+    targetUrl.searchParams.set("partner-id", partnerId);
+    targetUrl.searchParams.delete("class-id");
+  }
+  window.location.assign(targetUrl);
+};
+
+
+// Token exchange is not yet implemented with the embed webapp and not required in case of most of the partners
+const tokenRequestCallBack = () => {}
+
+GSU.createIframe({
+  targetElementId: 'iframe-target',
+  targetPage: 'discover',
+  partnerId: 'gsudemo',
+  navigationCallBack,
+  tokenRequestCallBack,
+  // This targetUrls object isn't necessary in production integrations, but it allows us to point to the dev environment.
+  targetUrls: {
+    discover: 'https://embed-webapp.www.gsudevelopment.com/discovery/{partnerId}',
+    watch: 'https://embed-webapp.www.gsudevelopment.com/watch/{partnerId}/{classId}',
+  },
+})
+```
+
+The content of the div will be replaced with the GetSetUp iframe.
+
+### Watch page
+
+Then on a second page where you want the user to watch the class:
+
+```html
+<div id="iframe-target"></div>
+```
+
+```ts
+import * as GSU from '@getsetup/embed'
+
+// This function is supplied by the hosting page.
+// This function will be called by the GSU iframe when it needs to navigate.
+const navigationCallBack = (
+  navigationAction: GSU.NavigationAction;
+  classId?: string;
+) => {
+  const targetUrl = new URL(window.location.href);
+  // The 'watch' navigation action is special because we need to pass the classId to the 'watch' page.
+  if (navigationAction == "watch" && classId) {
+    targetUrl.pathname = "/hostSitePath/watch";
+    targetUrl.searchParams.set("class-id", classId ?? "");
+  } else if (navigationAction == "discover") {
+    targetUrl.pathname = "/hostSitePath/discover";
+    targetUrl.searchParams.set("partner-id", partnerId);
+    targetUrl.searchParams.delete("class-id");
+  }
+  window.location.assign(targetUrl);
+};
+
+// Token exchange is not yet implemented with the embed webapp and not required in case of most of the partners
+const tokenRequestCallBack = () => {}
+
+//Get the sessionId from the query string.
+const urlParams = new URLSearchParams(window.location.search)
+const classId = urlParams.get('classId')
+
+GSU.createIframe({
+  targetElementId: 'iframe-target',
+  targetPage: 'watch',
+  classId: classId,
+  partnerId: 'gsudemo',
+  navigationCallBack,
+  tokenRequestCallBack,
+  // This targetUrls object isn't necessary in production integrations, but it allows us to point to the dev environment.
+  targetUrls: {
+    discover: 'https://embed-webapp.www.gsudevelopment.com/discovery/{partnerId}',
+    watch: 'https://embed-webapp.www.gsudevelopment.com/watch/{partnerId}/{classId}',
```

Comment:
These have to be mentioned separately in the docs too, 
prod URLs and dev URLs.

Reply:
Note: These are overridden URLs, used and specified only while testing locally or in a staging environment. 
We can skip these in the production, as the embed package by default points to the production ones.